native/acter/src/api/auth.rs

```diff
@@ -304,23 +305,29 @@ pub async fn register_under_config(
     RUNTIME
         .spawn(async move {
             let client = config.build().await?;
-            if let Err(resp) = client
-                .matrix_auth()
-                .register(register::v3::Request::new())
-                .await
-            {
-                // FIXME: do actually check the registration types...
-                if resp.as_uiaa_response().is_some() {
+            // user registration follows UIAA spec
+            // req-resp exchange continues until final stage of auth process
+            let request = register::v3::Request::new();
+            if let Err(e) = client.matrix_auth().register(request).await {
+                let Some(inf) = e.as_uiaa_response() else {
+                    bail!("No a uiaa response")
+                };
+                if let Some(err) = &inf.auth_error {
+                    bail!("Found auth error: {:?}", err.message);
+                }
```

Comment:
please note that this has recently be changed to support better error messages to users, please update the code here accordingly.

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1761/commits/4f1ee0e0417fc7e1ecc0119ff65aa79e6c6f8461